zwrap/sample_test.go

```diff
@@ -97,7 +97,7 @@ func TestSampler(t *testing.T) {
 		for i := 1; i < 10; i++ {
 			tt.logFunc(sampler, i)
 		}
-		expected := buildExpectation(tt.level, 1, 2, 5, 8)
+		expected := buildExpectation(tt.level, 1, 2, 4, 6, 8)
```

Comment:
For completeness, let's also add a test that passes an exact power-of-two value for `thereafter` (which ensures that we're handling the rounding-down logic properly).